content/6_Advanced/SPneg.mdx

```diff
@@ -214,7 +214,94 @@ int main() {
 ```
 
 </CPPSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class CyclesFinding {
+	static final int N = 2500;
+	static int n, m;
+	static int[] parent = new int[N + 1];
+	static long[] dist = new long[N + 1];
+	static List<Edge> graph = new ArrayList<>();
+	static Kattio io = new Kattio();
```

Comment:
i feel like a lot of these don't actually need to be static

Reply:
But without static keyword, those variables cann't be referenced from the static method (main).

Reply:
make them local to the function (except the ones the dfs uses obviously)

Reply:
fix it